subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/bean/AbstractNestedRuntimeBeanNode.java

```diff
@@ -104,6 +109,36 @@ public TaskDependencyContainer getTaskDependencies() {
                     }
                 };
             }
+            if (isMap()) {
+                return context -> {
+                    Object dependency = valueSupplier.get();
+                    if (dependency instanceof Map) {
+                        Map<Object, Object> map = (Map<Object, Object>) dependency;
+                        if (!map.isEmpty()) {
```

Comment:
💄 can we remove the duplicated collection handling?

Reply:
better?

Reply:
Yes, thank you :)